hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -43,19 +49,19 @@ class LowerTableIR(val typesToLower: DArrayLowering.Type) extends AnyVal {
   def lower(ir: IR): IR = ir match {
       case TableCount(tableIR) =>
         val stage = lower(tableIR)
-        invoke("sum", TInt64, stage.toIR(node => Cast(ArrayLen(ToArray(node)), TInt64)))
+        stage.wrapInBindings(invoke("sum", TInt64, stage.toIR(node => Cast(ArrayLen(ToArray(node)), TInt64))))
 
       case TableGetGlobals(child) =>
         val stage = lower(child)
-        GetField(stage.broadcastVals, stage.globalsField)
+        stage.wrapInBindings(GetField(stage.broadcastVals, stage.globalsField))
 
       case TableCollect(child) =>
         val lowered = lower(child)
         val elt = genUID()
         val cda = lowered.toIR(x => ToArray(x))
-        MakeStruct(FastIndexedSeq(
+        lowered.wrapInBindings(MakeStruct(FastIndexedSeq(
```

Comment:
TableCollect has an IR duplication problem as well, it seems. Putting everything in the bindings and forcing the globals/broadcastvals to just be refs will fix this.

(Don't fix that here)

Reply:
yep.

I'm thinking you could actually restructure the bindings/broadcast vals/globals to look something like:
```
  letBindings: IndexedSeq[(String, IR)],
  broadcastVals: IndexedSeq[String], // or IndexedSeq[Ref]
  globalsField: String, // or Ref
```
so that `letBindings` is the only place you actually create new bindings, and broadcastVals is just the subset of bindings you need to broadcast. This changes the structure of a reference to a broadcast value a tiny bit---instead of calling `GetField(broadcastVals, name)` you just have `Ref(name, typ)`, and you just end up needing to wrap the body with nested
```
Let(name, GetField(Ref(cdaGlobalName, _), name), …)
```

This kind of reflects how we use the broadcast values in the lowerer, anyways (as a binding env, not as a monolithic IR). You don't have to make that change here, but I think it's pretty small and if it's easier to do here I'm happy to review it.